compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -391,10 +391,13 @@ public void visit(BLangXMLNSStatement xmlnsStmtNode, AnalyzerData data) {
     @Override
     public void visit(BLangResourceFunction funcNode, AnalyzerData data) {
         visit((BLangFunction) funcNode, data);
+        BType returnType = funcNode.returnTypeNode.getBType();
+        checkForClientObjectTypeOrFunctionType(returnType, Types.getReferredType(returnType),
+                funcNode.returnTypeNode.getPosition());
```

Comment:
```ballerina
type CustomType0 float|int[]|CustomType3|int|string;
type CustomType1 float|int[]|CustomType3|int|string;
type CustomType3 client object{};
type CustomType4 CustomType3 | CustomType1;

client object{} & readonly successClient = client object {

};

public function testErrorInResourceMethodReturningObjectType() {
    var successClientObject = client object {
        resource function get clientReturnPath() returns CustomType4 {
            return successClient;
        }
    };
}
```
With this change the above program compiles fine. But it should give an error. right?

Reply:
Yes. Will check it

Reply:
Solved 2725ba33d9230d4c98b3b79818080c3ba31e21b3